course_info/templates/course_info/partials/people.html

```diff
@@ -10,200 +10,247 @@
+    </div>
+  </div>
+  <!-- begin add person expand-o-widget -->
 
-      <h3>People in {{title}}</h3>
+  <!-- begin success alerts -->
+  <uib-alert ng-repeat="person in successes" type="success"
+             close="closeAlert('successes', $index)" class="col-md-12">
+    <p><strong>{{ person.searchTerm }}</strong> was just {{ person.action }} this course:</p>
+    <div style="border-style: solid; border-width: 1px 0px 1px 0px; border-color: #ccc; padding:.5em; margin-top:1em;">
+      {{ person.profile.name_last }}, {{ person.profile.name_first }}
+      <badge role="{{ person.profile.role_type_cd }}"></badge> {{ person.profile.univ_id }}
+      <strong>Enrolled as {{ person.role.role_name }}</strong>
+    </div>
+  </uib-alert>
+  <!-- end success alerts -->
 
-      <uib-alert ng-repeat="person in successes" type="success"
-                                                 close="closeAlert('successes', $index)" class="col-md-12">
-        <p><strong>{{ person.searchTerm }}</strong> was just added to this course:</p>
-        <div style="border-style: solid; border-width: 1px 0px 1px 0px; border-color: #ccc; padding:.5em; margin-top:1em;">
-          {{ person.profile.name_last }}, {{ person.profile.name_first }}
-          <badge role="{{ person.profile.role_type_cd }}"></badge> {{ person.profile.univ_id }}
-          <strong>Enrolled as {{ person.role.role_name }}</strong>
-        </div>
-      </uib-alert>
+  <!-- begin add partial failure warning alerts -->
+  <uib-alert ng-repeat="partial in addPartialFailures" type="warning"
+             close="closeAlert('addPartialFailures', $index)" class="col-md-12">
+    <p>While adding <strong>{{ partial.searchTerm }}</strong> to this course:</p>
+    <p>{{ partial.text }}</p>
+  </uib-alert>
+  <!-- end add partial failure warning alerts -->
 
-      <!-- partial failure warning alerts -->
-      <!-- TODO - remove these once the API is rolling back on canvas failure -->
-      <uib-alert ng-repeat="partial in partialFailures" type="warning"
-                                                        close="closeAlert('partialFailures', $index)" class="col-md-12">
-        <p>While adding <strong>{{ partial.searchTerm }}</strong> to this course:</p>
-        <p>{{ partial.text }}</p>
-      </uib-alert>
+  <!-- begin remove failure warning alerts -->
+  <uib-alert ng-repeat="failure in removeFailures" type="warning"
+             close="closeAlert('removeFailures', $index)" class="col-md-12">
+    <div ng-switch on="failure.type">
+      <div ng-switch-when="noSuchUser">
+        <strong>{{ failure.profile.name_last }}, {{ failure.profile.name_first }}</strong>
+        has already been removed from this course.
+      </div>
+      <div ng-switch-when="noSuchCourse">
+        <strong>{{ failure.profile.name_last }}, {{ failure.profile.name_first }}</strong>
+        cannot be removed from this course because it no longer exists.
+      </div>
+      <div ng-switch-when="serverError">
```

Comment:
they've all got slightly different error messages, not sure how fall-through would work.